### PR TITLE
Network cleanup

### PR DESCRIPTION
**Pull request by nerzhul**
This PR intend to cleanup and make the network part a little bit more clear and reduce compilation times by splitting connection.h big header and socket.h header (isolate Socket from Address object)

* Move peerhandler to a specific header to reduce compilation times
* Move socket.{cpp,h} to network folder
* Move Address object to network/address.{cpp,h}
* Move network exceptions to network/networkexceptions.h
* Client: use unique_ptr for Connection
* Server/ClientIface: use shared_ptr for Connection

### CODE REVIEW

**Comment by nerzhul** [comment]
LINT is not okay, those files were ignored by blacklist but are perfectly compliant with our rules, i pass the LINT on them when PR is validated

File: src/network/socket.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	~UDPSocket();
+	void Bind(Address addr);
+
+	bool init(bool ipv6, bool noExceptions = false);
+
+	//void Close();
+	//bool IsOpen();
+	void Send(const Address & destination, const void * data, int size);
+	// Returns -1 if there is no data
+	int Receive(Address & sender, void * data, int size);
```
Comment (SmallJoker) [inline src/network/socket.h]: `void *data`, in `Send` too.

File: src/network/socket.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	UDPSocket() = default;
+
+	UDPSocket(bool ipv6);
+	~UDPSocket();
+	void Bind(Address addr);
+
+	bool init(bool ipv6, bool noExceptions = false);
+
+	//void Close();
+	//bool IsOpen();
```
Comment (SmallJoker) [inline src/network/socket.h]: Deprecated comments. Functions not existent.

**Comment by nerzhul** [comment]
Don't comment code style as i said i will run clang format -------- Message original --------Objet : Re: [minetest/minetest] Network cleanup (#6302)De : SmallJoker À : minetest/minetest Cc : Loïc Blot ,Author @SmallJoker commented on this pull request.



In src/network/socket.h:
> +void sockets_init();
+void sockets_cleanup();
+
+class UDPSocket
+{
+public:
+	UDPSocket() = default;
+
+	UDPSocket(bool ipv6);
+	~UDPSocket();
+	void Bind(Address addr);
+
+	bool init(bool ipv6, bool noExceptions = false);
+
+	//void Close();
+	//bool IsOpen();

Deprecated comments. Functions not existent.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/minetest/minetest","title":"minetest/minetest","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/minetest/minetest"}},"updates":{"snippets":[{"icon":"PERSON","message":"@SmallJoker commented on #6302"}],"action":{"name":"View Pull Request","url":"https://github.com/minetest/minetest/pull/6302#pullrequestreview-58144995"}}}

File: src/network/address.cpp [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+			if(i < 14)
+				os << ":";
+		}
+		return os.str();
+	}
+	else
+		return std::string("");
+#else
+	char str[INET6_ADDRSTRLEN];
+	if (inet_ntop(m_addr_family, (m_addr_family == AF_INET) ? (void*)&(m_address.ipv4.sin_addr) : (void*)&(m_address.ipv6.sin6_addr), str, INET6_ADDRSTRLEN) == NULL) {
```
Comment (SmallJoker) [inline src/network/address.cpp]: Line too long.

File: src/network/address.cpp [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+		u32 addr;
+		addr = ntohl(m_address.ipv4.sin_addr.s_addr);
+		a = (addr & 0xFF000000) >> 24;
+		b = (addr & 0x00FF0000) >> 16;
+		c = (addr & 0x0000FF00) >> 8;
+		d = (addr & 0x000000FF);
+		return itos(a) + "." + itos(b) + "." + itos(c) + "." + itos(d);
+	}
+	else if(m_addr_family == AF_INET6)
+	{
```
Comment (SmallJoker) [inline src/network/address.cpp]: Various bracket and spacing style errors. Would you be so nice and make this section look good as well?

**Pull request merged**